src/status/status.factory.js

```diff
@@ -0,0 +1,209 @@
+const { stripIndent } = require('common-tags');
+
+
+module.exports = (logProvider, stateProvider, logger) => {
+    //Helper functions
+    function activeMembers(events) {
+        for(var i=0; i < event.length; i++){
+            var obj = events.logs[i];
+                if(obj.action == "Participated"){
+                    return ({obj})
+                }
+            }
+    }
+
+    function activeEvents(events){
+     for(var i=0; i < events.team_id.length; i++){
+         var obj = events;
+         if(obj.halted == true){
+                return ({obj});
+            }
+        }
+    }
+
+    function happenedInPeriodUser(date, action, back){
+        const alerts = ({
+            notified:0,
+            missed:0
+        });
+        const settingDate = new Date();
+        settingDate.setDate(date);
+        if( back != 'all')
+        {
+            const hasHappened = settingDate - back;
```

Comment:
can you subtract dates like this? Since we are not sure how dates are represented you can pass in a dependency for subtracting dates and comparing them.